### PR TITLE
fix(Scripts/Commands): correct reloading creature_template table

### LINKED ISSUE #4991: Issue with Reload commands [$20]
##### CURRENT BEHAVIOUR:
-When using the reload creature_template command, server will close.
-.reload smart_scripts doesn't seem to do anything

##### EXPECTED BLIZZLIKE BEHAVIOUR:
-don't crash
-smart script should update

##### STEPS TO REPRODUCE THE PROBLEM:
1. find a creature, get his entry ID, do the command, see if it crashes
2. change a creature's SAI, use the sai reload command and see if it updates

##### EXTRA NOTES:
Seems like a lot of .reload commands are broken

##### AC HASH/COMMIT:
83e430f28a25b33175569749a0aec46ccc92c9b6

##### OPERATING SYSTEM:
windows 10

##### CUSTOMIZATIONS:

<bountysource-plugin>

---
There is a **[$20 open bounty](https://www.bountysource.com/issues/97357180-issue-with-reload-commands?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Changes Proposed:
- Correct reloading `creature_template` table

## Issues Addressed:
- Closes #4991

## Target Branch(es):
- [x] Master
 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Winfidonarleyan** [comment]
Test in Windows
image

File: src/server/game/Globals/ObjectMgr.cpp [hunk 1]
```diff
@@ -593,6 +519,84 @@ void ObjectMgr::LoadCreatureTemplates()
...
+    creatureTemplate.trainer_class     = uint32(fields[36].GetUInt8());
+    creatureTemplate.trainer_race      = uint32(fields[37].GetUInt8());
+    creatureTemplate.type              = uint32(fields[38].GetUInt8());
+    creatureTemplate.type_flags        = fields[39].GetUInt32();
+    creatureTemplate.lootid            = fields[40].GetUInt32();
+    creatureTemplate.pickpocketLootId  = fields[41].GetUInt32();
+    creatureTemplate.SkinLootId        = fields[42].GetUInt32();
+
+    for (uint8 i = SPELL_SCHOOL_HOLY; i < MAX_SPELL_SCHOOL; ++i)
+        creatureTemplate.resistance[i] = 0;
```
Comment (FrancescoBorzi) [inline src/server/game/Globals/ObjectMgr.cpp:575]: brackets please

File: src/server/game/Globals/ObjectMgr.cpp [hunk 2]
```diff
@@ -593,6 +519,84 @@ void ObjectMgr::LoadCreatureTemplates()
...
+
+    for (uint8 i = 0; i < MAX_CREATURE_SPELLS; ++i)
+        creatureTemplate.spells[i] = 0;
```
Comment (FrancescoBorzi) [inline src/server/game/Globals/ObjectMgr.cpp:578]: brackets please

**Comment by r0m1ntik** [comment]
Красавчик 💯 tested

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**